packages/python-editor/src/KernelDropdown.tsx

```diff
@@ -83,12 +84,13 @@ class DropDown extends React.Component<DropDownProps, DropDownState> {
    * Function: Creates drop down options with available python kernel specs.
    */
   private createOptionElems = (specs: KernelSpec.ISpecModels): void => {
-    const kernelNames: string[] = Object.keys(specs.kernelspecs);
-    kernelNames.forEach((specName: string, i: number) => {
+    const kernelSpecs: ISpecModel[] = Object.values(specs.kernelspecs);
+    kernelSpecs.forEach((spec: ISpecModel, i: number) => {
+      const displayName = spec.display_name;
       const elem = React.createElement(
         'option',
-        { key: i, value: specName },
-        specName
+        { key: i, value: displayName },
+        displayName
```

Comment:
Just to be clear, we should `display` the `displayName`, but continue using the `specName` as the value retrieved and associated during kernel creation.